tests/unit/contentproviders/test_swhid.py

```diff
@@ -0,0 +1,154 @@
+import json
+import os
+import io
+import tarfile
+import shutil
+import re
+import urllib
+import pytest
+import tempfile
+import logging
+import requests_mock
+
+from os import makedirs
+from os.path import join
+from unittest.mock import patch, MagicMock, mock_open
+from zipfile import ZipFile
+
+from repo2docker.contentproviders.swhid import Swhid, parse_swhid
+from repo2docker.contentproviders.base import ContentProviderException
+
+
+# this is a slightly striiped down copy of swh.model.cli.swhid_of_dir().
+# We do not use this later to prevent having to depend on swh.model[cli]
+def swhid_of_dir(path):
+    object = Directory.from_disk(path=path).get_data()
+    return swhid(DIRECTORY, object)
+
+
+def test_content_id():
+    swhid = Swhid()
+    assert swhid.content_id is None
+
+
+swhids_ok = [
+    "swh:1:dir:" + "0" * 40,
+    "swh:1:rev:" + "0" * 40,
+]
+swhids_ko = [
```

Comment:
What does the `_ko` stand for? First I thought it was a typo version of `_ok` but I think it is on purpose :)

Reply:
ko like knock-out, but I can use _nok is you prefer

Reply:
I don't love either `ko` or `nok`. I think for both I'd appreciate a comment "ko like knock-out, as in not ok" . Or we use a longer name that spells it out for simple minded people like me :D

Reply:
ok, changed to `swhids_invalid`